### PR TITLE
changes made

### PR DESCRIPTION
**Pull request by abrolnalin**

### CODE REVIEW

File: fixtures/numberpoolTollfreeResponse.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    "meta": {
+        "limit": 20,
+        "next": "",
+        "offset": 0,
+        "previous": "",
+        "total_count": 1
+    },
+    "objects": [
+        {
+            "account_phone_number_resource": "/v1/Account/MAODZKMDFJMJU3MTEYNG/Number/15799140348/",
```
Comment (narayana-plivo) [inline fixtures/numberpoolTollfreeResponse.json]: try to avoid the auth id to expose

File: powerpacks.go [hunk 2]
```diff
@@ -50,15 +50,31 @@ type ShortCode struct {
 	Added_on         string `json:"added_on,omitempty"`
 }
 
+type Tollfree struct {
+	Number_pool_uuid string `json:"number_pool_uuid,omitempty"`
```
Comment (narayana-plivo) [inline powerpacks.go]: under_score variable names are valid, they are not idiomatic. Follow the standard

File: powerpacks.go [hunk 3]
```diff
@@ -300,6 +372,18 @@ func (service *PowerpackService) Find_shortcode(shortcode string) (response *Fin
 	return
 }
 
+func (service *PowerpackService) Find_tollfree(tollfree string) (response *FindTollfreeResponse, err error) {
+	numberpool_path := service.Powerpack.NumberPoolUUID
```
Comment (narayana-plivo) [inline powerpacks.go]: it should be numberpoolUUID right, not numberpool_path

File: powerpacks.go [hunk 4]
```diff
@@ -300,6 +372,18 @@ func (service *PowerpackService) Find_shortcode(shortcode string) (response *Fin
 	return
 }
 
+func (service *PowerpackService) Find_tollfree(tollfree string) (response *FindTollfreeResponse, err error) {
```
Comment (oliver006) [inline powerpacks.go]: This function (and all other exported functions in this file)  should not be using `_` but be called e.g.`FindTollfree`.

File: powerpacks.go [hunk 5]
```diff
@@ -277,6 +313,30 @@ func (service *PowerpackService) Remove_number(number string, param NumberRemove
...
+		return
+	}
+	response = &TollfreeDeleteResponse{}
+	err = service.client.ExecuteRequest(req, response)
+	return
+}
+
+func (service *PowerpackService) Remove_shortcode(shortcode string) (response *ShortcodeDeleteResponse, err error) {
+	numberpool_path := service.Powerpack.NumberPoolUUID
+	uriSegments := strings.Split(numberpool_path, "/")
```
Comment (oliver006) [inline powerpacks.go]: You should validate that `uriSegments` has (at least) 6 elements before accessing `uriSegments[5]`.

File: powerpacks.go [hunk 6]
```diff
@@ -50,15 +50,31 @@ type ShortCode struct {
 	Added_on         string `json:"added_on,omitempty"`
 }
 
+type Tollfree struct {
```
Comment (oliver006) [inline powerpacks.go]: You might want to rename this to something less generic and more scoped to this package, e.g. `PowerpackTollFreeBase`

File: powerpacks.go [hunk 7]
```diff
@@ -50,15 +50,31 @@ type ShortCode struct {
...
+	Objects []Tollfree `json:"objects" url:"objects"`
+}
 type FindShortCodeResponse struct {
 	ApiID string `json:"api_id,omitempty"`
 	ShortCode
 	Error string `json:"error,omitempty"`
 }
+type FindTollfreeResponse struct {
+	ApiID string `json:"api_id,omitempty"`
+	Tollfree
```
Comment (oliver006) [inline powerpacks.go]: nit: move this to top of the struct

File: powerpacks.go [hunk 8]
```diff
@@ -50,15 +50,31 @@ type ShortCode struct {
...
 	BaseListPPKResponse
 	Objects []ShortCode `json:"objects" url:"objects"`
 }
+type TollfreeResponse struct {
+	BaseListPPKResponse
+	Objects []Tollfree `json:"objects" url:"objects"`
+}
 type FindShortCodeResponse struct {
 	ApiID string `json:"api_id,omitempty"`
 	ShortCode
```
Comment (oliver006) [inline powerpacks.go]: nit: move this to top of the struct

File: powerpacks.go (hunk 3 above)
Comment (abrolnalin) [reply powerpacks.go]: it was named numberpool_path in existing apis so i kept the same name

File: powerpacks.go (hunk 2 above)
Comment (abrolnalin) [reply powerpacks.go]: Existing structs for longcodes and shortcode uses number_pool_uuid in their structs so should i leave it like this only to maintain uniformity or should i remove the underscore

File: fixtures/numberpoolTollfreeResponse.json (hunk 1 above)
Comment (abrolnalin) [reply fixtures/numberpoolTollfreeResponse.json]: yes, will change

**Pull request merged**